### DISCUSSION TITLE
Dificultades al hacer push en la branch act2

### DISCUSSION THREAD

**Discussion by IsabelMD** [Taller-Abierto-de-Humanidades-Digitales/curso_python_extenso#17] [Dudas del curso]
**Hola, Jairo.** 

Cómo estás? Una ves más yo por aquí  con dificultades. 

Sucede que no he podido hacer el push al Branch act2.  
No sé, si se debe a que comencé haciendo las actividades en el Branch creado la semana pasada (IsabelMD) y no en el act2 que se solicita en el documento de la Actividad 2. Una vez me di cuenta de mi error, un poco tarde he de decir, porque ya no me faltaba sino el push; decido hacer un merge con IsabelMD y hay si hacer el push. Sin embargo, unque me dejo agregar los archivos, y el commit a la hora de hacer el push me sale el siguiente error: 

          isabelmd@Isabels-MacBook-Pro curso_python_extenso % git branch 
            IsabelMD
          * act2
            main
          isabelmd@Isabels-MacBook-Pro curso_python_extenso % git merge IsabelMD
          Already up to date.
          isabelmd@Isabels-MacBook-Pro curso_python_extenso % git add .
          isabelmd@Isabels-MacBook-Pro curso_python_extenso % git commit -m "Entrega de la actividad 2"
          [act2 ca9337f] Entrega de la actividad 2
           9 files changed, 0 insertions(+), 0 deletions(-)
           create mode 100644 .DS_Store
           create mode 100644 estudiantes/.DS_Store
           create mode 100644 estudiantes/Saidy Isabel Mosquera Duque/.DS_Store
           delete mode 100644 estudiantes/Saidy Isabel Mosquera Duque/Actividad_2/__pycache__/act2Validator.cpython-311.pyc
           rename estudiantes/Saidy Isabel Mosquera Duque/{Actividad_2 => }/act2.py (100%)
           rename estudiantes/Saidy Isabel Mosquera Duque/{Actividad_2 => }/act2Validator.py (100%)
           rename estudiantes/Saidy Isabel Mosquera Duque/{Actividad_2 => }/act2_2.py (100%)
           rename estudiantes/Saidy Isabel Mosquera Duque/{Actividad_2 => }/act2_3.py (100%)
           rename estudiantes/Saidy Isabel Mosquera Duque/{Actividad_2 => }/act2_4.py (100%)
          isabelmd@Isabels-MacBook-Pro curso_python_extenso % git push
          fatal: The current branch act2 has no upstream branch.
          To push the current branch and set the remote as upstream, use
          
              git push --set-upstream origin act2
          
          isabelmd@Isabels-MacBook-Pro curso_python_extenso % 

**Muchas gracias por la ayuda**

**Comment by gaborszabo17** [comment]
Yo también presento el mismo error :(
Aunque, cuando coloqué `git push --set-upstream origin act2 ` me funcionó, añado la imagen abajo

![image](https://user-images.githubusercontent.com/123516687/216985910-728d01f2-a7fb-44d0-b4ff-43b2c5c2654a.png)

**Comment by jairomelo** [comment]
Hola a ambos, 
El problema en este caso es más sencillo de resolver, solamente hay que escribir el push de esta manera:

 `git push -u origin act2` 

La solución que propone git es correcta, pero con el comando de arriba será más sencillo que lo apliquen para cualquier brazo que creen en el futuro. 

@Taller-Abierto-de-Humanidades-Digitales/fundamentos-programacion
